src/sonic-bgpcfgd/bgpcfgd/managers_advertise_rt.py

```diff
@@ -18,82 +22,105 @@ def __init__(self, common_objs, db, table):
             db,
             table,
         )
-        
+
         self.directory.subscribe([("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, "localhost/bgp_asn"),], self.on_bgp_asn_change)
         self.advertised_routes = dict()
 
 
-    OP_DELETE = 'DELETE'
-    OP_ADD = 'ADD'
-
+    OP_DELETE = "DELETE"
+    OP_ADD = "ADD"
 
     def set_handler(self, key, data):
+        log_debug("AdvertiseRouteMgr:: set handler")
+        if not self.__set_handler_validate(key, data):
+            return True
         vrf, ip_prefix = self.split_key(key)
-        self.add_route_advertisement(vrf, ip_prefix)
+        self.add_route_advertisement(vrf, ip_prefix, data)
 
         return True
 
-
     def del_handler(self, key):
+        log_debug("AdvertiseRouteMgr:: del handler")
         vrf, ip_prefix = self.split_key(key)
         self.remove_route_advertisement(vrf, ip_prefix)
 
-
-    def add_route_advertisement(self, vrf, ip_prefix):
+    def __set_handler_validate(self, key, data):
+        if data:
```

Comment:
If data is not present, we need to advertise without a community string (for backward compatibility). Can you please confirm.?

Reply:
@prsunny confirmed, the check after the 'or' is for the case which doesn't have the community string. 
if ("profile" in data and data["profile"] in ROUTE_MAPS) or data == {"":""}: